velox/common/memory/Memory.h

```diff
@@ -738,8 +738,12 @@ int64_t MemoryPoolImpl<Allocator, ALIGNMENT>::getMaxBytes() const {
 template <typename Allocator, uint16_t ALIGNMENT>
 void MemoryPoolImpl<Allocator, ALIGNMENT>::setMemoryUsageTracker(
     const std::shared_ptr<MemoryUsageTracker>& tracker) {
+  const auto currentBytes = getCurrentBytes();
+  if (memoryUsageTracker_) {
+    memoryUsageTracker_->update(-currentBytes);
+  }
   memoryUsageTracker_ = tracker;
-  memoryUsageTracker_->update(getCurrentBytes());
```

Comment:
Not sure if we want to do this in first place.  The tracker should be only used to track the usage delta, no? @HuamengJiang @oerling @tanjialiang

Reply:
The thing is, if we attach a tracker to a nonempty memory pool, it'll then result in undercounting until you know when to detach the tracker just in time. This basically makes the memory capping meaningless.

Reply:
Should the tracker be pre-updated by the existing amount before attached to the memory pool?  If tracker is meant to be track the absolute value, then we probably should set it in the constructor and not allow to change it in the middle of the life cycle of the pool.

Reply:
@HuamengJiang Curious in which use case we need to repeatedly set the tracker and change the tracker during the life cycle?